src/main/java/com/converterapp/model/JsonDto.java

```diff
@@ -68,28 +57,43 @@ public JsonDto(XmlDto xml) {
                 break;
             }
         }
-        this.isArray = childrenXML.size() != 0 && tempIsArray && childrenXML.size() == xml.getChildren().size();
         for (XmlDto child : childrenXML) {
-            childrenJSON.add(new JsonDto(child, false));
+            childrenJSON.add(generateNotRootJsonDto(child));
         }
-        this.children.addAll(childrenJSON);
-        this.root = true;
+        return new JsonDto(xml.getTagName(), xml.getAttributes(),
+                xml.getContent() == null ? null : xml.getContent().trim(),
+                childrenJSON, true, xml.isElement(),
+                childrenXML.size() != 0 && tempIsArray && childrenXML.size() == xml.getChildren().size()
+        );
     }
 
-    public JsonDto(String tagName, String fileContent, final boolean root, final boolean onlyContent, final boolean isArray) {
-        super(tagName == null ? "element" : tagName, new HashMap<>(), onlyContent ? fileContent : null);
-        this.root = root;
-        this.onlyContent = onlyContent;
-        this.isArray = isArray;
-        this.children = new ArrayList<>();
+    public static JsonDto generateJsonDto(String tagName, String fileContent,
+                                          final boolean root, final boolean onlyContent, final boolean isArray) {
+        List<JsonDto> childrenJSON = new ArrayList<>();
         if (isArray) {
             if (fileContent != null && !fileContent.isBlank()) {
-                this.children.addAll(createArrayChildrenList(fileContent));
+                childrenJSON.addAll(createArrayChildrenList(fileContent));
             }
         }
+        return new JsonDto(tagName == null ? "element" : tagName,
```

Comment:
org.apache.commons.lang3.StringUtils.defaultIfBlank(tagName, ELEMENT)

Move hardcoded strings to AppConstants
String ELEMENT = "element";